api-server/src/server/boot/challenge.js

```diff
@@ -333,6 +341,56 @@ function backendChallengeCompleted(req, res, next) {
     .subscribe(() => {}, next);
 }
 
+function saveChallenge(req, res, next) {
+  const user = req.user;
+  const { id, challengeFiles: files = [] } = req.body;
+
+  const savableChallenges = getChallenges()
+    .filter(challenge => challenge.challengeType === 14)
+    .map(challenge => challenge.id);
```

Comment:
This is a moderately expensive operation. It should be run once on boot, not on every call to `saveChallenge`.